pkg/kapp/diff/change_set_with_versioned_rs.go

```diff
@@ -175,12 +175,7 @@ func (d ChangeSetWithVersionedRs) newAddChangeFromUpdateChange(
 	newRes ctlres.Resource, updateChange Change) Change {
 
 	// Use update's diffs but create a change for new resource
-	addChange := NewChangePrecalculated(nil, newRes, newRes)
-	// TODO private field access
-	addChange.op = ChangeOpAdd
-	addChange.configurableTextDiff = updateChange.ConfigurableTextDiff()
-	addChange.opsDiff = updateChange.OpsDiff()
-	return addChange
+	return NewChangePrecalculated(nil, newRes, newRes, ChangeOpAdd, updateChange.ConfigurableTextDiff(), updateChange.OpsDiff())
```

Comment:
Do you think a comment which elaborates on why this change was necessary will help?

Reply:
I think the TODO `// TODO private field access` explains why this is changed.

I feel that comments should be added to explain why something is done (for future reference) or should be done (TODOs) instead of why something is changed (as that is only relevant for the PR). But do let me know if the new change doesn't make sense and requires a comment.